File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/multiline/comment/direct/Java/Go/s283652523_commented.go
package main

import (
	"fmt"
)

const MOD int64 = 1000000007

func main() {
	var n int
	fmt.Scan(&n)
	a := make([]int, n)
	for i := 0; i < n; i++ {
		fmt.Scan(&a[i])
	}
	if n == 0 {
		fmt.Println(0)
		return
	}
	sum := int64(a[n-1])
	ans := int64(0)
	for i := n - 2; i >= 0; i-- {
		a_i := int64(a[i])
		productTerm := (sum * a_i) % MOD
		ans += productTerm
		ans %= MOD
		sum += a_i
		sum %= MOD
	}
	fmt.Println(ans % MOD)
}

// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/multiline/comment/direct/Java/C/s283652523_commented.c
#include <stdio.h>
#include <stdlib.h>

#define MOD 1000000007

int main() {
    int N;
    scanf("%d", &N);
    int *A = malloc(N * sizeof(int));
    for (int i = 0; i < N; i++) {
        scanf("%d", &A[i]);
    }
    long long ans = 0;
    long long sum = A[N-1];
    for (int i = N-2; i >= 0; i--) {
        long long temp = (sum * A[i]) % MOD;
        ans = (ans + temp) % MOD;
        sum = (sum + A[i]) % MOD;
    }
    free(A);
    printf("%lld\n", ans);
    return 0;
} // End of Code
